client/src/lib/supabase/middleware.ts

```diff
@@ -1,5 +1,51 @@
 import { createServerClient } from "@supabase/ssr";
 import { NextResponse, type NextRequest } from "next/server";
+import { userHasExistingProfile } from "@/lib/actions/profile_setup"
+
+//debug utility function 
+const DEBUG = false;
+export function debug(msg: string){
+  if (DEBUG){
+    console.log("DEBUG: ", msg);
+  }
+}
+
+debug("***** BEGIN middleware.ts ******");
+debug("middleware.ts is always invoked!!!");
+
+//allowed public paths
+const publicPaths = ["/","/login","/explore","/signup","/about"];
+
+//whitelists auth'd user paths
+const validUserPaths = ["/message", "/profile", "/thread", "/track", "/profile_setup"];
+
+function hasBasePath(pathname: string, basepaths: string[]) {
+  for (let i = 0; i < basepaths.length; i++) {
+    debug("pathname: " + pathname);
+    debug("basepaths[i]" + basepaths[i]);
+    if (pathname == "/")
+      return true 
+    if (basepaths[i] == "/") {continue}
+    if (pathname.indexOf(basepaths[i]) == 0) {
+      return true
+    }
+  }
+  return false
+}
+function isPublicPath(pathname: string){
```

Comment:
this is redundant with the hasBasePath function. remove this

Reply:
It could be considered redundant for publicPaths but is necessary for validUserPaths. Still working on the Set implementation, maybe that will resolve the use of my functions.